go/l2geth-exporter/l1contracts/main.go

```diff
@@ -0,0 +1,36 @@
+package l1contracts
+
+import (
+	"context"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/go/l2geth-exporter/bindings"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// CTC interacts with the OVM CTC contract
+type CTC struct {
+	Ctx     context.Context
```

Comment:
We generally want to avoid storing `Context` instances inside structs. Instead, pass the context into `GetTotalElements` below. This will let you specify a timeout for the each call to `GetTotalElements`, and prevent issues like the one we saw the other day where the exporter hung while trying to query the L1.

Reply:
Ah, makes sense. See if the update does what you expect.